external/storm-kafka/src/jvm/org/apache/storm/kafka/AvroScheme.java

```diff
@@ -0,0 +1,59 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.storm.kafka;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.util.List;
+
+import org.apache.avro.Schema;
+import org.apache.avro.generic.GenericContainer;
+import org.apache.storm.avro.DefaultDirectAvroSerializer;
+import org.apache.storm.avro.DirectAvroSerializer;
+import org.apache.storm.spout.Scheme;
+import org.apache.storm.tuple.Fields;
+import org.apache.storm.tuple.Values;
+import org.apache.storm.utils.Utils;
+
+public class AvroScheme implements Scheme {
+    public static final String AVRO_SCHEME_KEY = "avro";
+    DirectAvroSerializer serializer = new DefaultDirectAvroSerializer();
```

Comment:
Could this be more flexible, the way AbstractAvroSerializer is above?  I would see creating an AvroSchema and supplying a serializer for it to use based on how the avro data is being put onto the Kafka topic.

My original use case was to read avro data that had been serialized onto a kafka topic with a Confluent serializer.  My solution was to create a bolt that that read the raw data with rawscheme, deserialized the raw bytes into a GenericRecord, and then emitted the generic record.

Something like this would have been very nice if I could give it a ConfluentDeserializer via dependency injection or something similar.